### PR TITLE
Stitch returns form

### PR DESCRIPTION
**Pull request by mmarkusik**
I want to merge this change because it stitches the order return form with api 

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [ ] Number of API calls is optimized.
1. [ ] The changes are tested.
1. [ ] Data-test are added for new elements.
1. [ ] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://feature-refund-products.api.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1874 Stitch returns page](https://app.clickup.com/t/2549495/SALEOR-1874)

**Review by krzysztofwolski — APPROVED** [review]

File: src/orders/components/OrderRefundReturnAmount/OrderRefundReturnAmount.tsx [hunk 1]
```diff
@@ -240,14 +240,13 @@ const OrderRefundAmount: React.FC<OrderRefundAmountProps> = props => {
...
         <Button
-          type="submit"
           color="primary"
           variant="contained"
           fullWidth
           size="large"
           onClick={onRefund}
           className={classes.refundButton}
-          disabled={disabled || disableRefundButton}
+          disabled={false}
```
[the remark this answers is not in the document]
Comment (mmarkusik) [reply src/orders/components/OrderRefundReturnAmount/OrderRefundReturnAmount.tsx:249]: Actually yes. Noticed a bug here, when calculating the refund for items when the whole order was refunded - will be added in fixes pr along with some other stuff, so for now just allowed it for development

File: src/orders/views/OrderReturn/utils.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  return itemsQuantities.reduce((result, { value: quantity, id }) => {
+    if (!quantity) {
+      return result;
+    }
+
+    const shouldReplace = !!itemsToBeReplaced.find(getById(id))?.value;
+
+    return [
+      ...result,
+      ({ [idKey]: id, quantity, replace: shouldReplace } as unknown) as T
```
Comment (orzechdev) [inline src/orders/views/OrderReturn/utils.tsx:63]: Why is it treated as `unknown`? Returned objects have strictly defined type (has `[idKey]`, `quantity` and `replace` keys), while `unknown` allows to pass arbitrary generic T type what may cause that returned value is not consistent with T type. In other words typescript will not warn if you write `getParsedLineData<{ anything: any }>`, while it should, as `anything` doesn't exist in returned objects.

Maybe restricting allowed types to `getParsedLineData<T extends OrderReturnLineInput | OrderReturnFulfillmentLineInput>` may help?

Comment (mmarkusik) [reply src/orders/views/OrderReturn/utils.tsx:63]: Good point! Unfortunately the `uknown` conversion was still needed by ts, but indeed now when initialing `getParsedLineData<{foo: boolean}>` typescript throws an error.

**Review by orzechdev — APPROVED** [review]

**Pull request merged**